src/test/java/com/delivery/user/controller/UserControllerTest.java

```diff
@@ -34,172 +35,211 @@
+            .content(new Gson().toJson(userDto))
+            .contentType(MediaType.APPLICATION_JSON)
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(redirectedUrl("/users/1"))
+            .andExpect(status().isCreated())
+            .andDo(print());
+    }
+
+    @DisplayName("회원 가입 실패 테스트 - 값이 없는 빈 객체 전달하여 회원 가입 실패 테스트")
+    @Test
+    void signUpFail() throws Exception {
+        // given
+        UserDto userDto = new UserDto();
+
+        // when
+        // then
+        mockMvc.perform(post("/users")
+            .content(new Gson().toJson(userDto))
+            .contentType(MediaType.APPLICATION_JSON)
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isBadRequest())
+            .andDo(print());
+    }
+
+    @DisplayName("회원 가입 실패 테스트 - 같은 이메일을 가진 유저가 존재하여 실패 테스트")
+    @Test
+    void signUpFailWithExistsUser() throws Exception {
+        // given
+        UserDto userDto = UserDto.builder()
+            .email("whdudgns2654@naver.com")
+            .name("조영훈")
+            .password("asdqwe1234567!@#")
+            .phoneNumber("010-1234-1234")
+            .status(DataStatus.DEFAULT)
+            .build();
+
+        given(userService.existsByEmail(userDto.getEmail())).willReturn(true);
+
+        // when
+        // then
+        mockMvc.perform(post("/users")
+            .content(new Gson().toJson(userDto))
+            .contentType(MediaType.APPLICATION_JSON)
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isConflict())
+            .andDo(print());
+    }
+
+    @DisplayName("회원 조회 성공 테스트")
+    @Test
+    void findUser() throws Exception {
+        // given
+        User user = User.builder()
+            .id(1L)
+            .email("whdudgns2654@naver.com")
+            .name("조영훈")
+            .password("asdqwe1234567!@#")
+            .phoneNumber("010-1234-1234")
+            .status(DataStatus.DEFAULT)
+            .build();
+
+        given(userService.find(user.getId())).willReturn(user);
+
+        // when
+        // then
+        mockMvc.perform(get("/users/" + user.getId())
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andExpect(content().json(new Gson().toJson(user)))
+            .andDo(print());
+    }
+
+    @DisplayName("회원 조회 실패 테스트")
+    @Test
+    void findUserFail() {
+        // given
+        long userId = 1L;
+
+        given(userService.find(userId)).willThrow(NoSuchElementException.class);
+
+        // when
+        Throwable thrown = catchThrowable(() -> mockMvc.perform(get("/users/" + userId)
+            .accept(MediaType.APPLICATION_JSON)));
+
+        // then
+        assertThat(thrown.getCause()).isInstanceOf(NoSuchElementException.class);
+    }
+
+    @DisplayName("회원 삭제 성공 테스트")
+    @Test
+    void deleteUser() throws Exception {
+        // given
+        long userId = 1L;
+
+        doNothing().when(userService).delete(userId);
```

Comment:
Mockitod와 BDDMockito가 혼용될 이유는 없을거 같아용
아래와 같이 대체가 가능합니다
```suggestion
willDoNothing().given(userService).delete(userId);
```